fuel-core-interfaces/src/txpool.rs

```diff
@@ -26,61 +26,66 @@ pub trait TxPoolDb:
     }
 }
 
-/// Subscriber interface that receive inserted/removed events from txpool.
-/// Usually needed for libp2p broadcasting and can be used to notify users of new tx.
-#[async_trait]
-pub trait Subscriber: Send + Sync {
-    async fn inserted(&self, tx: Arc<Transaction>);
-
-    async fn inserted_on_block_revert(&self, tx: Arc<Transaction>);
-
-    async fn removed(&self, tx: Arc<Transaction>, error: &Error);
-}
-
-#[async_trait]
-pub trait TxPool: Send + Sync {
+#[derive(Debug)]
+pub enum TxPoolMpsc {
```

Comment:
Isn't it over-engineering converting functionalities that can trivially be functions into async channels? If we have many enum variants for the things that should actually be async, it might be confusing.

Reply:
having a trait or channel does not in essence change a lot in terms of the interface, that is why firstly I implemented this as `trait TxPool` and now I am changing it to channels as it seems that it will give us a little bit better handling for requests in more concurrent mode.
Sender channel can emulate ordinary function call with a wrapped structure, so the channel is hidden behind it: https://github.com/FuelLabs/fuel-core/compare/rakita/txpool_channels...rakita/txpool_sender